src/tracer.js

```diff
@@ -231,13 +231,24 @@ export default class Tracer {
             ctx.traceId = parent.traceId;
             ctx.spanId = Utils.getRandom64();
             ctx.parentId = parent.spanId;
-            ctx.flags = parent.flags;
-
             // reuse parent's baggage as we'll never change it
             ctx.baggage = parent.baggage;
 
-            parent.finalizeSampling();
-            ctx.finalizeSampling();
+            // Parent exists without a sampling decision
+            // An example of where this happens is in mobile clients where sampler parameters
+            // are not propagated
+            if ((parent.flags & constants.DEFERRED_SAMPLING_MASK)
+                === constants.DEFERRED_SAMPLING_MASK) {
+                ctx.flags = parent.flags ^ constants.DEFERRED_SAMPLING_MASK
```

Comment:
Agree about running the same thing multiple times, but as Yuri said, I think that we shouldn't be modifying the parent span because it was created elsewhere.